File: Python/SubStringCheck.py
def substring(text, pattern):
    n = len(text)
    m = len(pattern)
    for i in range(n-m):
        match = True
        for j in range(m):
            if text[i+j] != pattern[j]:
                match = False
                break
        if match:
            return True

    return False

if __name__ == "__main__":
    text = input("Enter the main text: ")
    pattern = input("Enter the pattern: ")

    if substring(text,pattern):
        print("The text contains the sub string")
    else:
        print("The text doesn't contains sub string")

File: Java/SubStringCheck.java
package Day3;
import java.util.*;
public class SubStringCheck {
    public static boolean issSubString(String text, String pattern){
        int n = text.length();
        int m = pattern.length();
        for(int i = 0; i < n - m; i++){
           boolean match = true;
            for(int j = 0; j < m; j++){
                if(text.charAt(i+j) != pattern.charAt(j)){
                    match = false;
                    break;
                }
            }
            if (match){
                return true;
            }
        }
        return false;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the Main String 1: ");
        String text = sc.nextLine();
        System.out.println("Enter the pattern: ");
        String pattern = sc.nextLine();

        if (issSubString(text, pattern)){
            System.out.println("The text contains the substring ");
        }
        else{
            System.out.println("The text doesn't contains the substring ");
        }

    }
}
